spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolverSettings.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.storage;
+
+/**
+ * Holds settings for use with the {@link GoogleStorageResource}. These settings are
+ * optional.
+ */
+public class GoogleStorageProtocolResolverSettings {
```

Comment:
Why do you still try to keep this class?
Why my suggestion in the mentioned commit doesn't work for?

OK. If you really would like to keep this `setting` intermediate class, let's consider to do `GcpStorageProperties extends GoogleStorageProtocolResolverSettings`!

Reply:
Sorry I forgot to explain. So without this settings class I could 

1. put GoogleStorageProtocolResolver(GSPR) as bean in StorageAutoConfiguration like you suggested, but at that point the bean method can't provide the beanFactory to the GSPR.
2. use GcpStorageProperties in GSPR, but we decided not to have the storage project depend on the starter-storage project. 
3. get the "auto-create-files" setting from Spring inside GSPR, but I thought it would be better to not have that name hard-coded

Of course please let me know an option I'm missing.

Reply:
I just saw your explanation further down and thanks much for clearing it up for me; that's definitely something new I learned today!

Reply:
Did we forget to delete this class?

Reply:
I think know. See here: https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolver.java#L79.

This is the point of collaboration between end-user and the target `GoogleStorageProtocolResolver` which is very specific to be instantiated as a bean in the target end-user configuration.

Reply:
Seems like it's required, but why didn't we simply add this field to `GcpStorageProperties` instead of having this extra class?

Reply:
@artembilan it sounds like a `GcpStorageProperties` bean could be loaded instead, if I'm not missing anything.

Reply:
Right, you are missing the fact that `GcpStorageProperties` is in the `starter` module. That one which is Spring Boot aware already. But this *bridge* we need here in pure Store module for better end-developer experience.

Reply:
If that's the case, then we should probably merge `GcpStorageProperties` and `GoogleStorageProtocolResolverSettings` when moving the properties classes to the main modules as part of #155.

Reply:
Well, your concern there about Spring Boot dependency is valid.
If there is anything changed since there, let me know.
But I'd definitely stick away from Spring Boot dependencies in the core modules.
That's not their responsibility to carry Boot feature. True.

If there is an idea to merge something together, I would suggest `auto-configure` module and indeed keep starters as just dependency management. The fact is that even Spring Boot per se has all the auto-configuration just only in one module: https://github.com/spring-projects/spring-boot/tree/master/spring-boot-project/spring-boot-autoconfigure